### PR TITLE
Re-layout of Players Overlay

### PR DESCRIPTION
**Pull request by mousewax**
Added  instructions and inactive player separator to room.

Added CSS to make layout more useful and place your name at the top and active players above inactive players.

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: http://212.47.248.129:3761/pr-test (or any other room on that server)

**Comment by bjalder26** [comment]
I think it looks nice, and provides a nice new feature (separating out users playerName) .  However, I'm reluctant to be the one to approve this since I've never worked on UI stuff before and am horrible at CSS.
image

**Review by 96LawDawg — APPROVED** [review]
Well, I will approve it.  I know the least about coding, but ... (1) This is a big improvement in appearance. (2) Wish I had this before because it would have fixed a few issues in games I was playing with inexperienced VTT users.  (3) I want to encourage @mousewax to keep making improvements.

I looked at the CSS and although I don't understand it, I didn't see anything that jumped out at me as potentially a problem.  It also passed all the TestCafe checks, so should be safe to move to production.  However, I think someone else should do the merge just so there is a double check.

**Comment by mousewax** [comment]
In the interest of making sure everyone's comfortable, here's a run down of what I did:

The biggest change I made was moving the layout to flexbox rather than just a list of DIVs. In addition to the instructions under the h1, I added an extra separation node to the html.

This allowed me to apply a flexbox "order" to the various classes: myPlayerEntry (order: 1;), then regular divs (order: 2;) that are shortened and get stacked left to right, then the separation node (order: 3;) that's full width to break the active from inactive groups and finally the inactivePlayerEntry (order: 4;) which are likewise shortened to stack left to right.

The "me:" and "other players" titles were added as ::before and ::after pseudo-elements on the myPlayerEntry to keep the markup as simple as possible. 

Lastly, I added some styling to the color input to add the icon and remove the webkit styling.

Because I could order these in CSS, I didn't have to make any changes to the JS that builds the page or change any actual functionality.

**Comment by RaphaelAlvez** [comment]
hey I saw that you used WebKit.
This normally creates problems with firefox so I checked and it looks pretty bad.

It doesn't need to be identical or as good and in chrome/safari but maybe in this case a fix is needed. Maybe make the square smaller?

image

**Comment by mousewax** [comment]
I will make sure it is the same in Firefox! Thanks for checking it.

File: client/css/overlays/players.css [hunk 1]
```diff
@@ -1,23 +1,68 @@
...
+
+.teamColor::-webkit-color-swatch {
+  border-radius: 100%;
+  border: 1px solid rgba(255, 255, 255, 0.25);
+  /* background properties broken up to avoid erasing the selected colors */
+  background-image: url(/i/icons-white/user-solid-svg.svg);
+  background-repeat: no-repeat;
+  background-position: center center;
+  background-size: 1rem;
 }
```
Comment (RaphaelAlvez) [inline client/css/overlays/players.css:66]: Webkit does not work on firefox

**Review by RaphaelAlvez — CHANGES_REQUESTED** [review]
Just take a look at this. Is there anything to do about it?

looks great on safari and chrome. in general its a good improvement

**Comment by mousewax** [comment]
Gecko does not add the extraneous styling that webkit’s does and *generally
*doesn’t need those extra bits so there’s nothing to do for Firefox there.
But I definitely need to override other things on Firefox.

**Comment by mousewax** [comment]
Gecko now adds extraneous styling and pseudo elements. I just committed the styles to cover for it. Gecko and WebKit now look identical.

image

**Comment by bjalder26** [comment]
They look virtually identical to me (and pretty amazing).
Click update branch -> then I think RaphaelAlvez needs to resolve the change request -> then wait 24 hrs before merging.

File: client/css/overlays/players.css (hunk 1 above)
Comment (mousewax) [reply client/css/overlays/players.css]: The appropriate -moz equivalents were added, which I didn't knew were things. Usually it's a struggle getting around all of Apple's ornamentation. I look forward to a day where we get rid of vendor prefixes.

**Pull request merged**